### PR TITLE
Add UI for APK downloads

### PR DESCRIPTION
**Pull request by tbodt**

### CODE REVIEW

**Review by saagarjha — COMMENTED** [review]
Preliminary comments

File: app/APKFilesystem.m [hunk 1]
```diff
@@ -72,6 +73,18 @@ static int apkfs_fstat(struct fd *fd, struct statbuf *stat) {
...
+@interface DownloadCompletion : NSObject {
+    @public
+    BOOL complete;
+    NSError *error;
+    lock_t complete_lock;
+    cond_t complete_cond;
+}
+@end
+@implementation DownloadCompletion
+@end
```
[the remark this answers is not in the document]
Comment (tbodt) [reply app/APKFilesystem.m:86]: This is an object so I can have a weak reference to it

File: app/APKDownloadsViewController.m [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+@property (weak, nonatomic) IBOutlet NSLayoutConstraint *offscreenConstraint;
+
+@end
+
+@implementation APKDownloadsViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.downloads = [NSMutableArray new];
+    self.onscreen = NO;
```
[the remark this answers is not in the document]
Comment (tbodt) [reply app/APKDownloadsViewController.m:31]: No longer

File: app/APKDownloadsViewController.m [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    [self.downloads.firstObject.progress cancel];
+}
+
+- (void)setOnscreen:(BOOL)onscreen {
+    _onscreen = onscreen;
+    self.topConstraint.active = self.offscreenConstraint.active = NO;
+    self.topConstraint.active = onscreen;
+    self.offscreenConstraint.active = !onscreen;
+    [self.view layoutSubviews];
+}
```
[the remark this answers is not in the document]
Comment (tbodt) [reply app/APKDownloadsViewController.m]: What would you replace it with

File: app/APKFilesystem.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (saagarjha) [reply app/APKFilesystem.m:86]: Why do you need a weak reference?

File: app/APKDownloadsViewController.m (hunk 3 above)
[the remark this answers is not in the document]
Comment (saagarjha) [reply app/APKDownloadsViewController.m]: I'd fold it into the other method.

**Pull request merged**